modules/javafx.graphics/src/main/native-glass/mac/GlassMTLOffscreen.m

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (c) 2024, 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#import "GlassMTLOffscreen.h"
+
+//#define VERBOSE
+#ifndef VERBOSE
+    #define LOG(MSG, ...)
+#else
+    #define LOG(MSG, ...) GLASS_LOG(MSG, ## __VA_ARGS__);
+#endif
+
+@implementation GlassMTLOffscreen
+
+static NSArray *allModes = nil;
+
+- (id)initWithContext:(id<MTLDevice>)device
+         commandQueue:(id<MTLCommandQueue>)commandQueue
+          andIsSwPipe:(BOOL)isSwPipe
+{
+    self = [super init];
+    if (self != nil)
+    {
+        self->_fbo = [[GlassMTLFrameBufferObject alloc] init];
+        // if (self->_fbo == nil)
+        // {
+        //     TODO: implement PBuffer if needed
+        //     self->_fbo = [[GlassPBuffer alloc] init];
+        // }
+        [(GlassMTLFrameBufferObject*)self->_fbo setIsSwPipe:(BOOL)isSwPipe];
+        if (allModes == nil) {
+            allModes = [[NSArray arrayWithObjects:NSDefaultRunLoopMode,
+                                              NSEventTrackingRunLoopMode,
+                                              NSModalPanelRunLoopMode, nil] retain];
+        }
+        self->offScreenCommandQueue = commandQueue;
+        self->mtlDevice = device;
+    }
+    return self;
+}
+
+- (void)dealloc
+{
+    [(NSObject*)self->_fbo release];
+    self->_fbo = nil;
+    [super dealloc];
```

Comment:
here and elsewhere: should we protect against repeated calls to dealloc?  like `if(self->_fbo != nil)`